src/declaration.c

```diff
@@ -1653,6 +1653,7 @@ void VarDeclaration::setFieldOffset(AggregateDeclaration *ad, unsigned *poffset,
             {
                 const char *s = (t->ty == Tsarray) ? "static array of " : "";
                 ad->error("cannot have field %s with %ssame struct type", toChars(), s);
+                return;
```

Comment:
My comment was just for the PR reviewers to explain the mechanism of issue behavior.
Once bug fixed, the added return statement will be a trivial error handing. Any other dmd code don't have verbose comments for error handling, so I'm just followed current style.
Furthermore, the comment (== explanation of bug 13236 mechanism) will explain non-existent behavior after the bugfix. I don't think it will be helpful.